pkg/openstack/loadbalancer_test.go

```diff
@@ -534,3 +534,158 @@ func Test_getListenerProtocol(t *testing.T) {
 		})
 	}
 }
+
+func TestLbaasV2_checkListenerPorts(t *testing.T) {
+	type fields struct {
+		LoadBalancer LoadBalancer
+	}
+	type args struct {
+		service            *corev1.Service
+		curListenerMapping map[listenerKey]*listeners.Listener
+		isLBOwner          bool
+		lbName             string
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "error is not thrown if loadblanacer matches & if port is already in use by a lb",
+			args: args{
+				service: &corev1.Service{
+					Spec: corev1.ServiceSpec{
+						Ports: []corev1.ServicePort{
+							{
+								Name:     "service",
+								Protocol: "https",
+								Port:     9090,
+							},
+						},
+					},
+				},
+				curListenerMapping: map[listenerKey]*listeners.Listener{
+					{
+						Protocol: "https",
+						Port:     9090,
+					}: {
+						ID:   "listenerid",
+						Tags: []string{"test-lb", "test-lb2"},
+					},
+				},
+				isLBOwner: false,
+				lbName:    "test-lb",
+			},
+			wantErr: false,
+			fields: fields{
+				LoadBalancer: LoadBalancer{},
+			},
+		},
+		{
+			name: "error is thrown if loadbalancer doesn't matches & if port is already in use by a service",
+			args: args{
+				service: &corev1.Service{
+					Spec: corev1.ServiceSpec{
+						Ports: []corev1.ServicePort{
+							{
+								Name:     "service",
+								Protocol: "https",
+								Port:     9090,
+							},
+						},
+					},
+				},
+				curListenerMapping: map[listenerKey]*listeners.Listener{
+					{
+						Protocol: "https",
+						Port:     9090,
+					}: {
+						ID:   "listenerid",
+						Tags: []string{"test-lb", "test-lb1"},
+					},
+				},
+				isLBOwner: false,
+				lbName:    "test-lb2",
+			},
+			wantErr: true,
+			fields: fields{
+				LoadBalancer: LoadBalancer{},
+			},
+		},
+		{
+			name: "error is not thrown if lbOwner is present & no tags on service",
+			args: args{
+				service: &corev1.Service{
+					Spec: corev1.ServiceSpec{
+						Ports: []corev1.ServicePort{
+							{
+								Name:     "service",
+								Protocol: "https",
+								Port:     9090,
+							},
+						},
+					},
+				},
+				curListenerMapping: map[listenerKey]*listeners.Listener{
+					{
+						Protocol: "https",
+						Port:     9090,
+					}: {
+						ID: "listenerid",
+					},
+				},
+				isLBOwner: true,
+				lbName:    "test-lb",
+			},
+			wantErr: false,
+			fields: fields{
+				LoadBalancer: LoadBalancer{},
+			},
+		},
```

Comment:
We should have a case when `lbOwner=true` and there are some tags on the Service.

Reply:
done